README_BRANCHING.md

```diff
@@ -0,0 +1,41 @@
+This document attempts to capture the details of our branching strategy
+for Puppet Server.  This information is up-to-date as of 2015-03-06.
+
+## THE MOST IMPORTANT THING TO KNOW
+
+tl;dr: we are putting a freeze on merging up changes from `stable` to `master`
+until Puppet Server 2.0 ships.
```

Comment:
One thing I've been wondering and I think is good to get down in writing.

Why are we explicitly preventing merge-ups instead of resolving the conflicts we want to avoid?  That is to say, if we don't want to pull up a new dependency, why not do the merge-up as normal, undo the dependency change in the master branch, then in the future update the dependency (or not) as it makes sense for master?

This would allow us to continue to merge-up frequently, as normal, without pulling up the undesirable changes we don't want to pull up.